### PR TITLE
fix: fix applyPatch when patch and patchType is nil

### PR DESCRIPTION
**Pull request by CharlesQQ**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes https://github.com/karmada-io/karmada/issues/1703

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```
`karmada-controller-manager`: Fixed return err in case of interpreter webhook returns nil patch and nil patchType.

```

### CODE REVIEW

**Comment by XiShanYongYe-Chang** [comment]
Hi @CharlesQQ, do you mean when `patch` and `patchType` are nil, we shouldn't treat it as an error?

**Comment by CharlesQQ** [comment]
> do you mean when patch and patchType are nil, we shouldn't treat it as an error?

because following code not treat is as an error

https://github.com/karmada-io/karmada/blob/778b2eaa3b300d80b30d22cb598cd09dca0ea84d/pkg/resourceinterpreter/customizedinterpreter/webhook/resourceinterpretercontext.go#L132-L134

https://github.com/karmada-io/karmada/blob/778b2eaa3b300d80b30d22cb598cd09dca0ea84d/pkg/webhook/interpreter/response.go#L26-L45

**Review by XiShanYongYe-Chang — COMMENTED** [review]
Ok, I agree with you.

/lgtm

**Comment by XiShanYongYe-Chang** [comment]
@CharlesQQ, can you help add the release notes.

The affects conponent should be `karmada-controller-manager`.

**Comment by CharlesQQ** [comment]
> can you help add the release notes?

Is that ok?

**Comment by XiShanYongYe-Chang** [comment]
`The affects conponent should be karmada-controller-manager.`  Sorry, I didn't make it clear that this sentence is only an explanation, not the release-note.

**Comment by CharlesQQ** [comment]
> Sorry, I didn't make it clear that this sentence is only an explanation, not the release-note.

got it!

**Comment by XiShanYongYe-Chang** [comment]
Thanks~

File: pkg/resourceinterpreter/customizedinterpreter/customized.go [hunk 1]
```diff
@@ -251,6 +251,10 @@ func (e *CustomizedInterpreter) callHook(ctx context.Context, hook configmanager
 
 // applyPatch uses patchType mode to patch object.
 func applyPatch(object *unstructured.Unstructured, patch []byte, patchType configv1alpha1.PatchType) (*unstructured.Unstructured, error) {
+	if len(patch) == 0 && len(patchType) == 0 {
+		klog.Infof(" (%s: %s), patch and patchType is nil, give up apply patch ", object.GroupVersionKind().String(), object.GetName())
+		return object, nil
+	}
```
Comment (RainbowMango) [inline pkg/resourceinterpreter/customizedinterpreter/customized.go:257]:
```diff
diff --git a/pkg/resourceinterpreter/customizedinterpreter/customized.go b/pkg/resourceinterpreter/customizedinterpreter/customized.go
index 35af4a5b..05e9aed2 100644
--- a/pkg/resourceinterpreter/customizedinterpreter/customized.go
+++ b/pkg/resourceinterpreter/customizedinterpreter/customized.go
@@ -252,7 +252,7 @@ func (e *CustomizedInterpreter) callHook(ctx context.Context, hook configmanager
 // applyPatch uses patchType mode to patch object.
 func applyPatch(object *unstructured.Unstructured, patch []byte, patchType configv1alpha1.PatchType) (*unstructured.Unstructured, error) {
        if len(patch) == 0 && len(patchType) == 0 {
-               klog.Infof(" (%s: %s), patch and patchType is nil, give up apply patch ", object.GroupVersionKind().String(), object.GetName())
+               klog.Infof("Skip apply patch for object(%s: %s) as patch and patchType is nil", object.GroupVersionKind().String(), object.GetName())
                return object, nil
        }
        switch patchType {

```
I think `skip` would be better than `give up`

Comment (CharlesQQ) [reply pkg/resourceinterpreter/customizedinterpreter/customized.go]: > I think skip would be better than give up

good suggestion

**Review by RainbowMango — APPROVED** [review]
/lgtm
/approve

**Pull request merged**